### ISSUE TITLE
Unclear usage or "env.chronics_handler.reset" in the documentation

### ISSUE THREAD

**Issue by Mleyliabadi** [rte-france/Grid2Op#169]
## Environment
 - Grid2op version: `1.4.0`
 - System: Linux-4.4.0-189-generic-x86_64-with-Ubuntu-18.04-bionic`
 - Additional system information

## Bug description

I would like to resample the chronics and to use each resampled set to test the performance of different proxies. There is a functionality called «shuffle» under the environment's «chronics_handler» class which aims at performing this resampling task. However, when I use the shuffle functionality and I reset the environment, despite the condition defined in shuffle, it returns only the first available chronic. 

### Command line

```bash
# command line used if any 
```

### Code snippet

```python
from leap_net.proxy.utils import create_env
env_name="l2rpn_case14_sandbox"
use_lightsim_if_available = True
env = create_env(env_name, use_lightsim_if_available=use_lightsim_if_available)
# sampling with replacement five chronics with identifiers varied from 990 to 1000
kept_before_reset = env.chronics_handler.shuffle(lambda x: np.random.choice(np.arange(990,1000), size=5, replace=True))
kept_after_reset = kept = env.chronics_handler.real_data.reset()
print('Chronics kept before environment reset\n', kept_before_reset)
print('Chronics kept after environment reset\n', kept_after_reset)
```

## Current output

```
Chronics kept before environment reset
 ['/home/tfuser/data_grid2op/l2rpn_case14_sandbox/chronics/0999'
 '/home/tfuser/data_grid2op/l2rpn_case14_sandbox/chronics/0998'
 '/home/tfuser/data_grid2op/l2rpn_case14_sandbox/chronics/0996'
 '/home/tfuser/data_grid2op/l2rpn_case14_sandbox/chronics/0993'
 '/home/tfuser/data_grid2op/l2rpn_case14_sandbox/chronics/0991']
Chronics kept after environment reset
 ['/home/tfuser/data_grid2op/l2rpn_case14_sandbox/chronics/0000']

```

## Expected output

```
Chronics kept before environment reset
 ['/home/tfuser/data_grid2op/l2rpn_case14_sandbox/chronics/0999'
 '/home/tfuser/data_grid2op/l2rpn_case14_sandbox/chronics/0998'
 '/home/tfuser/data_grid2op/l2rpn_case14_sandbox/chronics/0996'
 '/home/tfuser/data_grid2op/l2rpn_case14_sandbox/chronics/0993'
 '/home/tfuser/data_grid2op/l2rpn_case14_sandbox/chronics/0991']
Chronics kept after environment reset
['/home/tfuser/data_grid2op/l2rpn_case14_sandbox/chronics/0999'
 '/home/tfuser/data_grid2op/l2rpn_case14_sandbox/chronics/0998'
 '/home/tfuser/data_grid2op/l2rpn_case14_sandbox/chronics/0996'
 '/home/tfuser/data_grid2op/l2rpn_case14_sandbox/chronics/0993'
 '/home/tfuser/data_grid2op/l2rpn_case14_sandbox/chronics/0991']

```

**Comment by BDonnot** [comment]
Hello,
I am not sure "shuffle" is the right tool for that purpose. Shuffles aims at... shuffling the data, and not selecting which to keep and which not to keep.

Grid2op comes with different tools to precisely use only some chronics:
- there is a function to perform the "train / val" split: https://grid2op.readthedocs.io/en/latest/environment.html#grid2op.Environment.Environment.train_val_split_random
- there is a function to explicitly select a specific chronics: https://grid2op.readthedocs.io/en/latest/environment.html#study-always-the-same-chronics
- there is a function to sample chronics "at random" with weights for the chronics: https://grid2op.readthedocs.io/en/latest/environment.html#sampling-the-chronics

I will investigate this further, but definitely, shuffle is not the right tool to use for this purpose.

**Comment by BDonnot** [comment]
After looking at the documentation of `chronics_handler.reset` here : https://grid2op.readthedocs.io/en/latest/chronics.html#grid2op.Chronics.Multifolder.reset it says:
"""
**NB** This “reset” is different from the “env.reset”. It should be only called after the function to set the filtering function has been called.
"""
So this is a misusage of "env.chronics_handler.reset". This should not be called execpt in a very specific condition that is not met here. 

Did you mix it with "env.reset" ?

**Comment by Mleyliabadi** [comment]
Hi Benjamin, Thanks for your answer and the proposed functions to use. 

Concerning your second comment, I have used both env.reset() and env.chronics_handler.reset(), the results were same. After the reset, only the first chronic is maintained.  

The example provided in documentation concerning the usage of shuffle function leads me believe that this function is adapted for the resampling of chronics.

**Comment by BDonnot** [comment]
I totally agree there are lots of typos, some unclear explanation, or examples that are not working (first pages of the doc are approximately a year now, and grid2op evolved a lot). 

Yes there are issue with the doc, and thanks for pointing out something ill explained here (i'll clarify it asap)

However, in this case:
- there are more suited alternative to achieve what you want to do, more efficient, and less error prone (3 alternatives in this case). Please condiser using them instead.
- the doc states that you should not call `env.chronics_handler.reset` in this setting (cf. my second message). So please don't use it if you continue with this non efficient strategy.
- I agree the example in the doc can be misleading as there are some better ways, with other function to do what the example says. It is however working as intended :blush:  See a working example below:

```python
import numpy as np
import grid2op
np.random.seed(0)

env_name = "l2rpn_case14_sandbox"
env = grid2op.make(env_name)
# sampling with replacement five chronics with identifiers varied from 990 to 1000
chron_id_to_keep = np.random.choice(np.arange(990, 1000), size=5, replace=True)
kept_before_reset = env.chronics_handler.shuffle(lambda x: chron_id_to_keep)
print('Chronics kept before environment reset\n', kept_before_reset)
for i in range(10):
    obs = env.reset()
    print(f"Path of the chronics used: {env.chronics_handler.data.path}")

print("Now i shuffle for a different order (same chronics)")
env.chronics_handler.shuffle()
for i in range(10):
    obs = env.reset()
    print(f"Path of the chronics used: {env.chronics_handler.data.path}")

print("Now i resample the chronics:")
chron_id_to_keep = np.random.choice(np.arange(990, 1000), size=5, replace=True)
kept_before_reset = env.chronics_handler.shuffle(lambda x: chron_id_to_keep)
print('New chronics kept: \n', kept_before_reset)
for i in range(10):
    obs = env.reset()
    print(f"Path of the chronics used: {env.chronics_handler.data.path}")
```

Gives for me:
```
Chronics kept before environment reset
 ['/home/benjamin/data_grid2op/l2rpn_case14_sandbox/chronics/0995'
 '/home/benjamin/data_grid2op/l2rpn_case14_sandbox/chronics/0990'
 '/home/benjamin/data_grid2op/l2rpn_case14_sandbox/chronics/0993'
 '/home/benjamin/data_grid2op/l2rpn_case14_sandbox/chronics/0993'
 '/home/benjamin/data_grid2op/l2rpn_case14_sandbox/chronics/0997']
Path of the chronics used: /home/benjamin/data_grid2op/l2rpn_case14_sandbox/chronics/0990
Path of the chronics used: /home/benjamin/data_grid2op/l2rpn_case14_sandbox/chronics/0993
Path of the chronics used: /home/benjamin/data_grid2op/l2rpn_case14_sandbox/chronics/0993
Path of the chronics used: /home/benjamin/data_grid2op/l2rpn_case14_sandbox/chronics/0997
Path of the chronics used: /home/benjamin/data_grid2op/l2rpn_case14_sandbox/chronics/0995
Path of the chronics used: /home/benjamin/data_grid2op/l2rpn_case14_sandbox/chronics/0990
Path of the chronics used: /home/benjamin/data_grid2op/l2rpn_case14_sandbox/chronics/0993
Path of the chronics used: /home/benjamin/data_grid2op/l2rpn_case14_sandbox/chronics/0993
Path of the chronics used: /home/benjamin/data_grid2op/l2rpn_case14_sandbox/chronics/0997
Path of the chronics used: /home/benjamin/data_grid2op/l2rpn_case14_sandbox/chronics/0995
Now i shuffle for a different order (same chronics)
Path of the chronics used: /home/benjamin/data_grid2op/l2rpn_case14_sandbox/chronics/0997
Path of the chronics used: /home/benjamin/data_grid2op/l2rpn_case14_sandbox/chronics/0990
Path of the chronics used: /home/benjamin/data_grid2op/l2rpn_case14_sandbox/chronics/0995
Path of the chronics used: /home/benjamin/data_grid2op/l2rpn_case14_sandbox/chronics/0993
Path of the chronics used: /home/benjamin/data_grid2op/l2rpn_case14_sandbox/chronics/0993
Path of the chronics used: /home/benjamin/data_grid2op/l2rpn_case14_sandbox/chronics/0997
Path of the chronics used: /home/benjamin/data_grid2op/l2rpn_case14_sandbox/chronics/0990
Path of the chronics used: /home/benjamin/data_grid2op/l2rpn_case14_sandbox/chronics/0995
Path of the chronics used: /home/benjamin/data_grid2op/l2rpn_case14_sandbox/chronics/0993
Path of the chronics used: /home/benjamin/data_grid2op/l2rpn_case14_sandbox/chronics/0993
Now i resample the chronics:
New chronics kept: 
 ['/home/benjamin/data_grid2op/l2rpn_case14_sandbox/chronics/0999'
 '/home/benjamin/data_grid2op/l2rpn_case14_sandbox/chronics/0993'
 '/home/benjamin/data_grid2op/l2rpn_case14_sandbox/chronics/0995'
 '/home/benjamin/data_grid2op/l2rpn_case14_sandbox/chronics/0992'
 '/home/benjamin/data_grid2op/l2rpn_case14_sandbox/chronics/0994']
Path of the chronics used: /home/benjamin/data_grid2op/l2rpn_case14_sandbox/chronics/0993
Path of the chronics used: /home/benjamin/data_grid2op/l2rpn_case14_sandbox/chronics/0995
Path of the chronics used: /home/benjamin/data_grid2op/l2rpn_case14_sandbox/chronics/0992
Path of the chronics used: /home/benjamin/data_grid2op/l2rpn_case14_sandbox/chronics/0994
Path of the chronics used: /home/benjamin/data_grid2op/l2rpn_case14_sandbox/chronics/0999
Path of the chronics used: /home/benjamin/data_grid2op/l2rpn_case14_sandbox/chronics/0993
Path of the chronics used: /home/benjamin/data_grid2op/l2rpn_case14_sandbox/chronics/0995
Path of the chronics used: /home/benjamin/data_grid2op/l2rpn_case14_sandbox/chronics/0992
Path of the chronics used: /home/benjamin/data_grid2op/l2rpn_case14_sandbox/chronics/0994
Path of the chronics used: /home/benjamin/data_grid2op/l2rpn_case14_sandbox/chronics/0999
```

Which is the desired behaviour.

Long story short: avoid using functions when the doc says otherwise :blush:

**Comment by BDonnot** [comment]
Fixed in release 1.5.0
